api/src/main/java/org/apache/iceberg/util/CharSequenceSet.java

```diff
@@ -54,10 +54,21 @@ public boolean isEmpty() {
     return wrapperSet.isEmpty();
   }
 
+  public CharSequenceWrapper wrapper() {
+    if (containsWrapper == null) {
+      synchronized (this) {
+        if (containsWrapper == null) {
+          this.containsWrapper = ThreadLocal.withInitial(() -> CharSequenceWrapper.wrap(null));
+        }
+      }
+    }
+    return containsWrapper.get();
+  }
+
   @Override
   public boolean contains(Object obj) {
     if (obj instanceof CharSequence) {
-      return wrapperSet.contains(containsWrapper.set((CharSequence) obj));
+      return wrapperSet.contains(wrapper().set((CharSequence) obj));
```

Comment:
should it be set back to `null` after the operation?

Reply:
Yeah, this is probably a good idea.